lisa/tools/gcc.py

```diff
@@ -26,11 +26,13 @@ def command(self) -> str:
     def can_install(self) -> bool:
         return True
 
-    def compile(self, filename: str, output_name: str = "") -> None:
+    def compile(
+        self, filename: str, output_name: str = "", arguments: str = ""
+    ) -> None:
         if output_name:
-            self.run(f"{filename} -o {output_name}")
+            self.run(f"{arguments} {filename} -o {output_name}", shell=True)
```

Comment:
1. It needs force_run arguments to make sure every time, it's recompiled.
2. It's going complex, so compose the command line, and call `run` once. Like,

```python
cmd = f"{arguments} {filename}"
if output_name:
    cmd += f"-o {output_name}"

self.run....
```

Reply:
updated